plugins/com.google.cloud.tools.eclipse.appengine.validation/src/com/google/cloud/tools/eclipse/appengine/validation/XmlValidator.java

```diff
@@ -89,7 +101,27 @@ void validate(IFile resource, byte[] bytes)
         }
       }
     } catch (SAXException ex) {
-      createSaxErrorMessage(resource, ex);
+      createSaxErrorMessage(resource, (SAXParseException) ex.getException());
+    }
+  }
+  
+  void xsdValidation(IResource resource) throws CoreException {
+    try {
+      String xsd = helper.getXsd();
+      if (xsd != null) {
+        URL xsdPath = AppEngineWebXmlValidator.class.getResource(xsd);
+        InputStream stream = xsdPath.openStream();
```

Comment:
You need to close the stream (preferable with the try-resources block), unless it is guaranteed that, e.g., `StreamSource` will always close it. Opening an input stream will consume a file descriptor in this case, and on systems where the number of open files are very limited (e.g., http://stackoverflow.com/questions/34588/how-do-i-change-the-number-of-open-files-limit-in-linux), OS may refuse to create any new file descriptor once it hits the limit.

Of course, the stream will be closed when it is garbage-collected, but you never know when the garbage collection will kick in. (It's basically the same principle that you shouldn't use `finalize()` almost all the time.)

I know we are a bit lenient about closing streams in tests, but that's because it is a short-lived process that will die soon in an isolated domain, and it s extremely unlikely that it will hit a limit. But for production applications, it's desirable to reclaim resources as soon as we can.